### ISSUE TITLE
@JsonSerialize/Deserialize without @JsonProperty generation?

### ISSUE THREAD

**Issue by mattclamp** [immutables/immutables#150]
Is this a feature? If not, can I ask that it is? 

I have detected modules that are supposed to use a certain Property Naming Strategy and the auto generation of `@JsonProperty` is overriding my settings. :(

**Comment by elucash** [comment]
This definitely gets in the way. I've tried to came up more reliable handling, though have no confidence that it good solution either. Thank you for reporting this issue. Could you please check latest snapshot if it makes things any better?

```xml
    <dependency>
      <groupId>org.immutables</groupId>
      <artifactId>value</artifactId>
      <version>2.1.0-SNAPSHOT</version>
    </dependency>
```

in

```xml
  <repositories>
    <repository>
      <id>sonatype-nexus-snapshots</id>
      <name>Sonatype Nexus Snapshots</name>
      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
    </repository>
  </repositories>
```

**Comment by mattclamp** [comment]
It seems to now not add those generated tags, but my fields are losing other annotations (not auto-generated) I had on them, which are also important.

**Comment by elucash** [comment]
Just have couple of questions:

1. Could you please try to use `@Value.Style(passAnnotations={MyAnnotation.class,..}..)` on value type (see [how to apply styles](immutables.github.io/style.html))
2. What kind of target element type is for missing annotation. Do they allow placement on fields or only methods?

This might help resolve the issue, we can make further fixes if we came up with best recipe for the annotation handling in such cases.

**Comment by ldriscoll** [comment]
is there any way that we can have the `passAnnotations` apply to the newly created `Json` object, since the Json object is used for the JsonCreator, and we have some annotations that need to apply to that?

**Comment by elucash** [comment]
linking #162 here

**Comment by elucash** [comment]
Closing as minimal missing functionality was provided in PRs #158 and #162. Released in 2.0.21. Thank you!
